src/frequenz/channels/_anycast.py

```diff
@@ -20,54 +20,165 @@
 
 
 class Anycast(Generic[_T]):
-    """A channel for sending data across async tasks.
-
-    Anycast channels support multiple senders and multiple receivers.  A message sent
-    through a sender will be received by exactly one receiver.
+    """A channel that delivers each message to exactly one receiver.
+
+    # Description
+
+    [Anycast][frequenz.channels.Anycast] channels support multiple
+    [senders][frequenz.channels.Sender] and multiple
+    [receivers][frequenz.channels.Receiver]. Each message sent through any of the
+    senders will be received by exactly one receiver.
+
+    <center>
+    ```bob
+    .---------. msg1                           msg1  .-----------.
+    | Sender  +------.                       .------>| Receiver  |
+    '---------'      |      .----------.     |       '-----------'
+                     +----->| Channel  +-----+
+    .---------.      |      '----------'     |       .-----------.
+    | Sender  +------'                       '------>| Receiver  |
+    '---------' msg2                           msg2  '-----------'
```

Comment:
Yeah, I thought about that, but I think if we graph all the possible combinations it can get pretty messy. The idea of this diagram was to quickly and briefly show how the channel works conceptually. Maybe we should clarify this in more visible way though (like a *IMPORTANT* admonition or something), but I think making the diagram more complicated will not help much.

Reply:
What do you think about this?

```diff
     messages in the channel's buffer. The channel's buffer size can be configured at
     creation time via the `limit` argument.
 
+    The first receiver that is awaited will get the next message. When multiple
+    receivers are waiting, the [asyncio][] loop scheduler picks a receiver for each next
+    massage.
+
+    This means that, in practice, there might be only one receiver receiving all the
+    messages, depending on how tasks are schduled.
+
+    If you need to ensure some delivery policy (like round-robin or uniformly random),
+    then you will have to implement it yourself.
+
     To create a new [senders][frequenz.channels.Sender] and
     [receivers][frequenz.channels.Receiver] you can use the
```

Reply:
About adding the caption, I didn't find a nice way to do it. [mkdocs-material captions](https://squidfunk.github.io/mkdocs-material/reference/images/#image-captions) seemed like a perfect fit, but having a *svgbob* diagram instead of an image is not supported (the code is not converted to an image) so it looks really ugly.

Then I tried with:

```markdown
    <center style="clear: right">
    ```bob
    .---------. msg1                           msg1  .-----------.
    | Sender  +------.                       .------>| Receiver  |
    '---------'      |      .----------.     |       '-----------'
                     +----->| Channel  +-----+
    .---------.      |      '----------'     |       .-----------.
    | Sender  +------'                       '------>| Receiver  |
    '---------' msg2                           msg2  '-----------'

    ```

    <span style="font-size: smaller">*Two receivers waiting for messages sent by two
    senders*</span>
    </center>
```

![image](https://github.com/frequenz-floss/frequenz-channels-python/assets/1031485/e8246d1c-a708-4f22-916f-94455c8a7b18)

Which looks nice, but adds a lot of syntactic noise to the text docs.

Finally I tried:

```markdown
    <center>
    ```bob
    .---------. msg1                           msg1  .-----------.
    | Sender  +------.                       .------>| Receiver  |
    '---------'      |      .----------.     |       '-----------'
                     +----->| Channel  +-----+
    .---------.      |      '----------'     |       .-----------.
    | Sender  +------'                       '------>| Receiver  |
    '---------' msg2                           msg2  '-----------'

       Two receivers waiting for messages sent by two senders
    ```
    </center>
```

![image](https://github.com/frequenz-floss/frequenz-channels-python/assets/1031485/18d26cd2-1d1e-4481-9cd0-e253a689964e)

Which is much more lightweight, but then it doesn't look that nice.

My impression is adding the caption doesn't really add much more info, what it says seems pretty obvious from the diagram, and if it is not, maybe I should just remove the diagram, the idea was the other way around, that the diagram helps to understand the text not that we need text to understand the diagram :laughing: 

If you really think this needs clarification or really helps to understand better, then I can go with either of those 2 alternatives, I leave that to you.